shapely/tests/test_flatcoords.py

```diff
@@ -0,0 +1,351 @@
+        *shapely.to_ragged_array([geom, geom], include_z=True)
+    )
+    assert_geometries_equal(actual, [geom, geom])
+
+
+@pytest.mark.parametrize("geom", all_types)
+def test_include_z(geom):
+    _, coords, _ = shapely.to_ragged_array([geom, geom], include_z=True)
+    # For 2D geoms, z coords are filled in with NaN
+    assert np.isnan(coords[:, 2]).all()
+
+
+@pytest.mark.parametrize("geom", all_types_3d)
+def test_include_z_false(geom):
+    _, coords, _ = shapely.to_ragged_array([geom, geom], include_z=False)
+    # For 3D geoms, z coords are dropped
+    assert coords.shape[1] == 2
+
+
+@pytest.mark.parametrize("geom", all_types_not_supported)
+def test_raise_geometry_type(geom):
+    with pytest.raises(ValueError):
+        shapely.to_ragged_array([geom, geom])
+
+
+def test_points():
+    arr = shapely.from_wkt(
+        [
+            "POINT (0 0)",
+            "POINT (1 1)",
+            "POINT EMPTY",
+            "POINT EMPTY",
+            "POINT (4 4)",
+            "POINT EMPTY",
+        ]
+    )
+    typ, result, offsets = shapely.to_ragged_array(arr)
+    expected = np.array(
+        [[0, 0], [1, 1], [np.nan, np.nan], [np.nan, np.nan], [4, 4], [np.nan, np.nan]]
+    )
+    assert typ == shapely.GeometryType.POINT
+    assert_allclose(result, expected)
+    assert len(offsets) == 0
+
+    geoms = shapely.from_ragged_array(typ, result)
+    assert_geometries_equal(geoms, arr)
+
+
+def test_linestrings():
+    arr = shapely.from_wkt(
+        [
+            "LINESTRING (30 10, 10 30, 40 40)",
+            "LINESTRING (40 40, 30 30, 40 20, 30 10)",
+            "LINESTRING EMPTY",
+            "LINESTRING EMPTY",
+            "LINESTRING (10 10, 20 20, 10 40)",
+            "LINESTRING EMPTY",
+        ]
+    )
+    typ, coords, offsets = shapely.to_ragged_array(arr)
+    expected = np.array(
+        [
+            [30.0, 10.0],
+            [10.0, 30.0],
+            [40.0, 40.0],
+            [40.0, 40.0],
+            [30.0, 30.0],
+            [40.0, 20.0],
+            [30.0, 10.0],
+            [10.0, 10.0],
+            [20.0, 20.0],
+            [10.0, 40.0],
+        ]
+    )
+    expected_offsets = np.array([0, 3, 7, 7, 7, 10, 10])
+    assert typ == shapely.GeometryType.LINESTRING
+    assert_allclose(coords, expected)
+    assert len(offsets) == 1
+    assert_allclose(offsets[0], expected_offsets)
+
+    result = shapely.from_ragged_array(typ, coords, offsets)
+    assert_geometries_equal(result, arr)
+
+
+def test_polygons():
+    arr = shapely.from_wkt(
+        [
+            "POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))",
+            "POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10), (20 30, 35 35, 30 20, 20 30))",
+            "POLYGON EMPTY",
+            "POLYGON EMPTY",
+            "POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))",
+            "POLYGON EMPTY",
+        ]
+    )
+    typ, coords, offsets = shapely.to_ragged_array(arr)
+    expected = np.array(
+        [
+            [30.0, 10.0],
```

Comment:
Trailing 0's can be removed, and suggest grouping these to make it easier to compare to the input geometries, e.g.,

```Python
expected = np.array(
    [
        # first polygon
        [30, 10], [40, 40], [20, 40], ...
        # second polygon (exterior ring, interior ring)
        [35, 10], ...
        [20, 30], ...
        # empty polygons omitted from coordinates
        # fifth polygon
        [30, 10], ...
```

Reply:
That won't work with black, but I suppose I can use comments to turn off/on formatting around those blocks